lib/src/Uncertainty/Algorithm/Sensitivity/SobolIndicesAlgorithmImplementation.cxx

```diff
@@ -913,7 +913,10 @@ void SobolIndicesAlgorithmImplementation::setDesign(const Sample & inputDesign,
     if (!(referenceVariance_[j] > 0.0))
       throw InvalidArgumentException(HERE) << "Null output sample variance";
 
+  // reset intermediate results
   alreadyComputedIndicesDistribution_ = false;
+  varianceI_.clear();
```

Comment:
You chose to clear `varianceI_` in SobolIndicesAlgorithmImplementation.cxx, which fixes the bug. But why not using `alreadyComputedIndicesDistribution_` instead of `varianceI_` at:
https://github.com/openturns/openturns/blob/186499f86b16a03b976b0d8b578f9903430d1fb7/lib/src/Uncertainty/Algorithm/Sensitivity/SobolIndicesAlgorithmImplementation.cxx#L131
to detect the change in the data? I think that it would make the intent much clearer.

Reply:
that's only for the distribution